[BUG] regex integration test failures on Databricks-11.3

**Describe the bug**
Below tests are failing due to mismatch in result between GPU and CPU runs.

```
AssertionError: GPU and CPU string values are different at [3, 'regexp_replace(a, [a\\d]{0,2}, x, 1)']
```
- [ ] test_regexp_replace_digit
- [ ] test_re_replace_repetition

**Steps/Code to reproduce bug**
Build and run integration tests on Databricks-11.3 environment.
Build  steps - /home/ubuntu/spark-rapids$ ./jenkins/databricks/build.sh

Test steps:  Modify the test.sh under jenkins/databrikcs to run a specific test.
                Example: Comment all lines starting from `export TEST_PARALLEL=${TEST_PARALLEL:-4}` in the file and add below lines:
            
               ```
                export TEST_PARALLEL=${TEST_PARALLEL:-1}
                TEST_NAME=$1
                bash /home/ubuntu/spark-rapids/integration_tests/run_pyspark_from_build.sh -k $TEST_NAME
               ```

Run Test -  `./jenkins/databricks/test.sh test_regexp_replace_digit `

**Expected behavior**
All above tests should pass on Databricks-11.3 shim

**Environment details (please complete the following information)**
Databricks-11.3 runtime.

**Additional context**
Add any other context about the problem here.

--- a/sql-plugin/src/main/330db/scala/com/nvidia/spark/rapids/shims/SparkShims.scala
+++ b/sql-plugin/src/main/330db/scala/com/nvidia/spark/rapids/shims/SparkShims.scala
@@ -60,6 +60,8 @@ object SparkShimImpl extends Spark321PlusDBShims {
 
   override def getExecs: Map[Class[_ <: SparkPlan], ExecRule[_ <: SparkPlan]] =
     super.getExecs ++ PythonMapInArrowExecShims.execs
+
+  override def reproduceEmptyStringBug: Boolean = false
 }
 
 trait ShimGetArrayStructFields extends ExtractValue {

